src/Gateways/TransitGateway.php

```diff
@@ -146,35 +218,49 @@ public function process_admin_options() {
 		$option_name = sprintf( 'woocommerce_%s_settings', $this->id );
 		$settings    = get_option( $option_name );
 
-		$this->user_id  = $settings['user_id'];
-		$this->password = $settings['password'];
+		if ( $this->is_production ) {
+			$this->user_id  = $settings['user_id'];
```

Comment:
Can we and should we use _$this->get_credential_setting_ here and where applicable below on the next few lines instead of accessing the array directly?

Reply:
I've investigated this and it seems that class properties are not set yet by the time this hook is called.